### PR TITLE
Sirusbr

### PR DESCRIPTION
**Pull request by sirus-ms**
## Description

[Provide a brief description of the changes made in this pull request.]

## Related Issues

[Reference any related issues or tasks that this pull request addresses.]

## Proposed Changes

[List the specific changes made in this pull request.]

## Checklist

- [ ] I have tested these changes locally.
- [ ] I have reviewed the code changes.
- [ ] I have updated the documentation (if necessary).
- [ ] I have added appropriate unit tests (if applicable).

## Additional Notes

[Add any additional notes or context that may be helpful for the reviewer(s).]

### CODE REVIEW

File: graphrag/index/__main__.py [hunk 1]
```diff
@@ -81,6 +81,12 @@
         help="Overlay default configuration values on a provided configuration file (--config).",
         action="store_true",
     )
+    parser.add_argument(
```
Comment (amritpalms) [inline graphrag/index/__main__.py:84]: Isn't it already part of the settings.yaml?

File: graphrag/index/cli.py [hunk 2]
```diff
@@ -106,8 +107,8 @@ def index_cli(
             ValueError("ContextId is invalid: It should be a valid Guid")
         if (context_operation != ContextSwitchType.Activate and context_operation != ContextSwitchType.Deactivate):
             ValueError("ContextOperation is invalid: It should be Active or DeActive")
-        graphrag_config = _read_config_parameters(root, config, progress_reporter)
-        _switch_context(graphrag_config, context_operation, context_id, progress_reporter)
```
Comment (amritpalms) [inline graphrag/index/cli.py:110]: nit: give space after commans in params

File: graphrag/index/context_switch/contextSwitcher.py [hunk 3]
```diff
@@ -1,22 +1,247 @@
...
+    def __init__(self, root_dir:str , config_dir:str,reporter: ProgressReporter,
+                 context_id:str, community_level:int ,
+                 data_dir: str = None,
+                 optimized_search: bool= False):
+
+        self.root_dir=root_dir
+        self.config_dir=config_dir
+        self.data_dir=data_dir
+        self.reporter=reporter
+        self.context_id=context_id
```
Comment (logomachic) [inline graphrag/index/context_switch/contextSwitcher.py:44]: Wouldn't it make more sense to have the context_id be part of activation? This is a new ContextSwitcher per context

File: graphrag/index/context_switch/contextSwitcher.py [hunk 4]
```diff
@@ -1,22 +1,247 @@
...
         #3. GraphDB: use cosmos db client to load data into Cosmos DB.
         #4. KustoDB: use Kusto client to load embedding data into Kusto.
 
-        return 0
+        data_dir=self.data_dir
+        root_dir=self.root_dir
+        config_dir=self.config_dir
+        reporter=self.reporter
+        context_id=self.context_id
+        optimized_search=self.optimized_search
```
Comment (amritpalms) [inline graphrag/index/context_switch/contextSwitcher.py:92]: we should get everything out related to kusto in separate module.
Something like below.
Activate:
GraphDb.Provision(contextId
KustoDb.Provision(ContextId)
For each path in File_paths:
      KustoDb.Ingest(path)
     GraphDb.Ingest(Path)

**Pull request merged**